python/ray/autoscaler/autoscaler.py

```diff
@@ -141,29 +142,32 @@ def _update(self):
 
         self.last_update_time = now
         nodes = self.workers()
+        managed_workers = self.managed_workers()
         # Check pending nodes immediately after fetching the number of running
         # nodes to minimize chance number of pending nodes changing after
         # additional nodes are launched.
         num_pending = self.pending_launches.value
         self.load_metrics.prune_active_ips(
             [self.provider.internal_ip(node_id) for node_id in nodes])
+        del nodes  # Always work in terms of managed_workers.
         target_workers = self.target_num_workers()
 
-        if len(nodes) >= target_workers:
+        if len(managed_workers) >= target_workers:
```

Comment:
Can you avoid such a big delta by retaining the `nodes` variable name? You can add another all_nodes to include unmanaged ones.